lib/pengine/tests/native/native_find_rsc_test.c

```diff
@@ -0,0 +1,350 @@
+group_rsc(void **state) {
+    pe_resource_t *exim_group = NULL;
+
+    /* Find the "exim-group" resource. */
+    for (GList *iter = data_set->resources; iter != NULL; iter = iter->next) {
+        pe_resource_t *rsc = (pe_resource_t *) iter->data;
+
+        if (strcmp(rsc->id, "exim-group") == 0) {
+            exim_group = rsc;
+            break;
+        }
+    }
+
+    assert_non_null(exim_group);
+
+    /* Passes because NULL was passed for node, regardless of flags. */
+    assert_ptr_equal(exim_group, native_find_rsc(exim_group, "exim-group", NULL, 0));
+    assert_ptr_equal(exim_group, native_find_rsc(exim_group, "exim-group", NULL, pe_find_current));
+
+    /* Fails because resource is not a clone. */
+    assert_null(native_find_rsc(exim_group, "exim-group", NULL, pe_find_clone));
+    assert_null(native_find_rsc(exim_group, "exim-group", cluster01, pe_find_clone));
+
+    /* Fails because none of exim-group's children are running on cluster01, even with the right flags. */
+    assert_null(native_find_rsc(exim_group, "exim-group", cluster01, pe_find_current));
+
+    /* Fails because pe_find_current is required if a node is given. */
+    assert_null(native_find_rsc(exim_group, "exim-group", cluster01, 0));
+
+    /* Passes because one of exim-group's children is running on cluster02. */
+    assert_ptr_equal(exim_group, native_find_rsc(exim_group, "exim-group", cluster02, pe_find_current));
+}
+
+static void
+inactive_group_rsc(void **state) {
+    pe_resource_t *inactive_group = NULL;
+
+    /* Find the "inactive-group" resource. */
+    for (GList *iter = data_set->resources; iter != NULL; iter = iter->next) {
+        pe_resource_t *rsc = (pe_resource_t *) iter->data;
+
+        if (strcmp(rsc->id, "inactive-group") == 0) {
+            inactive_group = rsc;
+            break;
+        }
+    }
+
+    assert_non_null(inactive_group);
+
+    /* Passes because NULL was passed for node, regardless of flags. */
+    assert_ptr_equal(inactive_group, native_find_rsc(inactive_group, "inactive-group", NULL, 0));
+    assert_ptr_equal(inactive_group, native_find_rsc(inactive_group, "inactive-group", NULL, pe_find_current));
+
+    /* Fails because resource is not a clone. */
+    assert_null(native_find_rsc(inactive_group, "inactive-group", NULL, pe_find_clone));
+    assert_null(native_find_rsc(inactive_group, "inactive-group", cluster01, pe_find_clone));
+
+    /* Fails because none of inactive-group's children are running. */
+    assert_null(native_find_rsc(inactive_group, "inactive-group", cluster01, pe_find_current));
+    assert_null(native_find_rsc(inactive_group, "inactive-group", cluster02, pe_find_current));
+
+    /* Passes because of flags. */
+    assert_ptr_equal(inactive_group, native_find_rsc(inactive_group, "inactive-group", cluster01, pe_find_inactive));
+    /* Passes because of flags. */
+    assert_ptr_equal(inactive_group, native_find_rsc(inactive_group, "inactive-group", cluster02, pe_find_inactive));
```

Comment:
for completeness, a NULL node with pe_find_inactive should work too

Reply:
Added.